src/utils.rs

```diff
@@ -0,0 +1,28 @@
+pub fn init_file_per_thread_logger() {
+    // There's a problem with borrow checker and static lifetimes across lambdas running
+    // in different threads, so we just use the global instead of function argument.
```

Comment:
Is this comment explaining why `LOG_FILENAME_PREFIX` is being used as a global here rather than as a function argument? If so, I don't think we need such a comment; it's fine to use global constants like `LOG_FILENAME_PREFIX`.